java/configuration/src/main/java/sleeper/configuration/properties/UserDefinedInstanceProperty.java

```diff
@@ -538,279 +540,286 @@ public interface UserDefinedInstanceProperty extends InstanceProperty {
-            .build();
+            .propertyGroup(PropertyGroup.COMPACTION).build();
     UserDefinedInstanceProperty COMPACTION_QUEUE_VISIBILITY_TIMEOUT_IN_SECONDS = named("sleeper.compaction.queue.visibility.timeout.seconds")
             .description("The visibility timeout for the queue of compaction jobs.")
             .defaultValue("900")
-            .build();
+            .propertyGroup(PropertyGroup.COMPACTION).build();
     UserDefinedInstanceProperty COMPACTION_KEEP_ALIVE_PERIOD_IN_SECONDS = named("sleeper.compaction.keepalive.period.seconds")
             .description("The frequency, in seconds, with which change message visibility requests are sent to extend the\n" +
                     "visibility of messages on the compaction job queue so that they are not processed by other processes.\n" +
                     "This should be less than the value of sleeper.queue.visibility.timeout.seconds.")
             .defaultValue("300")
-            .build();
+            .propertyGroup(PropertyGroup.COMPACTION).build();
     UserDefinedInstanceProperty COMPACTION_JOB_CREATION_LAMBDA_PERIOD_IN_MINUTES = named("sleeper.compaction.job.creation.period.minutes")
             .description("The rate at which the compaction job creation lambda runs (in minutes, must be >=1).")
             .defaultValue("1")
             .validationPredicate(Utils::isPositiveInteger)
-            .build();
+            .propertyGroup(PropertyGroup.COMPACTION).build();
     UserDefinedInstanceProperty COMPACTION_JOB_CREATION_LAMBDA_MEMORY_IN_MB = named("sleeper.compaction.job.creation.memory")
             .description("The amount of memory for the lambda that creates compaction jobs.")
             .defaultValue("1024")
-            .build();
+            .propertyGroup(PropertyGroup.COMPACTION).build();
     UserDefinedInstanceProperty COMPACTION_JOB_CREATION_LAMBDA_TIMEOUT_IN_SECONDS = named("sleeper.compaction.job.creation.timeout.seconds")
             .description("The timeout for the lambda that creates compaction jobs in seconds.")
             .defaultValue("900")
             .validationPredicate(Utils::isValidLambdaTimeout)
-            .build();
+            .propertyGroup(PropertyGroup.COMPACTION).build();
     UserDefinedInstanceProperty MAXIMUM_CONCURRENT_COMPACTION_TASKS = named("sleeper.compaction.max.concurrent.tasks")
             .description("The maximum number of concurrent compaction tasks to run.")
             .defaultValue("300")
-            .build();
+            .propertyGroup(PropertyGroup.COMPACTION).build();
     UserDefinedInstanceProperty COMPACTION_TASK_CREATION_PERIOD_IN_MINUTES = named("sleeper.compaction.task.creation.period.minutes")
             .description("The rate at which a check to see if compaction ECS tasks need to be created is made (in minutes, must be >= 1).")
             .defaultValue("1")
-            .build(); // >0
+            // >0
```

Comment:
This comment should probably be on the line with what it's referring to. Should there be a validation predicate?